technische-universitat-dortmund-ag-virtual-machining.csl

```diff
@@ -0,0 +1,141 @@
+  </macro>
+  <macro name="collection-title">
+    <text variable="collection-title"/>
+    <text variable="collection-number" prefix=" "/>
+  </macro>
+  <macro name="volume">
+    <text variable="volume" prefix=" "/>
+  </macro>
+  <macro name="date">
+    <date variable="issued">
+      <date-part name="year" prefix="(" suffix=")"/>
+    </date>
+  </macro>
+  <macro name="issue">
+    <choose>
+      <if type="article-journal article-newspaper article-magazine" match="any">
+        <text variable="issue" prefix=" "/>
+      </if>
+      <else-if type="chapter bill book graphic legal_case legislation motion_picture report song" match="any">
+	      <text variable="event-place" prefix=", "/>
+        <text macro="editor" prefix=", "/>
+	      <text macro="publisher" prefix=", "/>
+      </else-if>
+      <else>
+        <group prefix=". " delimiter=", ">
+          <choose>
+            <if type="thesis">
+              <text variable="genre" text-case="capitalize-first"/>
+            </if>
+          </choose>
+        </group>
+      </else>
+    </choose>
+  </macro>
+  <macro name="pages">
+	<text variable="page" prefix=": "/>
+  </macro>
+  <macro name="access">
+   	<group>
+		<choose>
+			<if  variable="ISBN" type="book chapter" match="none">
+				<text variable="DOI" prefix="DOI: "/>
+			</if>
+      <else-if variable="ISBN" match="any">
+			  <text variable="ISBN" prefix="ISBN "/>
+      </else-if>
+		</choose>
+	</group>
+  </macro>
+  <macro name="sort-date">
+    <date variable="issued">
+      <date-part name="day" suffix="."/>
+      <date-part name="month" suffix="."/>
+	  <date-part name="year"/>
+    </date>
+  </macro>
+  <macro name="editor">
+    <names variable="editor">
+      <name delimiter-precedes-last="never" initialize-with="." sort-separator=","/>
+      <label form="short" prefix=" (" suffix=")"/>
+    </names>
+  </macro>
+  <macro name="point-locators">
+    <choose>
+      <if variable="locator">
+        <choose>
+          <if locator="page" match="none">
+            <label variable="locator" form="short" suffix=" "/>
+          </if>
+        </choose>
+        <text variable="locator"/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=" ">
+      <text variable="publisher"/>
+    </group>
+  </macro>
+  <citation et-al-min="4" et-al-use-first="1" disambiguate-add-year-suffix="true" disambiguate-add-names="true" disambiguate-add-givenname="true" after-collapse-delimiter="; ">
+    <layout prefix="(" suffix=")" delimiter="; ">
+      <group delimiter=", ">
+        <group delimiter=" ">
+          <text macro="authors"/>
+          <text macro="date" />
+        </group>
+        <text macro="point-locators"/>
+      </group>
+    </layout>
+  </citation>
+  <bibliography hanging-indent="false" et-al-min="11" et-al-use-first="7" subsequent-author-substitute="---" entry-spacing="0">
+    <sort>
+      <key macro="sort-date" sort="descending"/>
```

Comment:
This seems very odd -- the bibliography is sorted by date, not by author names?
Also, the date as currently specified in the `date-sort` macro is day-month-year, leading to completely unpredictable sorting. Is that really what's expected?